includes/functions.php

```diff
@@ -2080,86 +2074,88 @@ function tsml_import_has_changes($data_source_url, $meetings, $data_source_count
-			if ( is_numeric( $dow )) {
-				$dayofweek = $days[date($dow)]; 
-			} else {
-				$dayofweek = $dow; 
-			}
+		$feed_slugs[$meeting['slug']] = $meeting_slug;
+
+		// has the meeting been updated since the last refresh?
+		$current_meeting_last_update = strtotime($meeting['updated']);
+		if ($current_meeting_last_update > $data_source_last_refresh) {
+
+			$meeting_name = $meeting['name'];
+			$meeting_update_date = date("M j, Y  g:i a", $current_meeting_last_update);
 			
 			if ($is_matched) {
-				$message_lines[] = "<tr style='color:gray;'><td>Change</td><td >$mtg_name</td><td>$dayofweek</td><td>$mtg_updte</td></tr>";
+				$message_lines[] = "<tr style='color:gray;'><td>Change</td><td >$meeting_name</td><td>$dayofweek</td><td>$meeting_update_date</td></tr>";
 			}
 			else {
-				$message_lines[] = "<tr style='color:green;'><td>Add New</td><td >$mtg_name</td><td>$dayofweek</td><td>$mtg_updte</td></tr>";
+				$message_lines[] = "<tr style='color:green;'><td>Add New</td><td >$meeting_name</td><td>$dayofweek</td><td>$meeting_update_date</td></tr>";
 			}
 		}
 	}
-
-	sort($feed_slugs);
-
-	// list removed meetings found in local db
+	
+	// list meetings in local database which are not matched with feed
 	foreach ($db_meetings as $db_meeting) {
 
-	    $mtg_slug =  $db_meeting['slug'];
-		if (is_array($db_meeting['day'])) {
-			$dow = implode("",  $db_meeting['day']);
-		} else {
-			$dow = $db_meeting['day'];
-		}
+		$dayofweek = $dow_number = '';
+		tsml_get_day_of_week_info( $db_meeting['day'], $dayofweek, $dow_number);
+	    $meeting_slug = $db_meeting['slug'];
 
-	    $is_matched = in_array( $mtg_slug, $feed_slugs );
+	    $is_matched = in_array( $meeting_slug, $feed_slugs );
 
 		if (!$is_matched) {
-		    $is_there_change = true;
-			if ( is_numeric( $dow )) {
-				$dayofweek = $days[date($dow)]; 
-			} else {
-				$dayofweek = $dow; 
-			}
-			$updated = $data_source_last_refresh;
-			$cur_meeting_lastupdate = strtotime($updated);
-			$mtg_updte = date("M j, Y  g:i a", $updated);
-			$mtg_name = $db_meeting['name'];
-			$message_lines[] = "<tr style='color:red;'><td>Remove</td><td >$mtg_name</td><td>$dayofweek</td><td>* $mtg_updte</td></tr>";
+			$meeting_update_date = date("M j, Y  g:i a", $data_source_last_refresh);
+			$meeting_name = $db_meeting['name'];
+			$message_lines[] = "<tr style='color:red;'><td>Remove</td><td >$meeting_name</td><td>$dayofweek</td><td>* $meeting_update_date</td></tr>";
 		}
 	}
+	
+	return $message_lines;
+}
+
+//function:	Returns corresponding day of week string and number for the day input
+if (!function_exists('tsml_get_day_of_week_info')) {
+	function tsml_get_day_of_week_info( $meeting_day_input, &$dayofweek, &$dow_number )
+	{
+		global $tsml_days;
+		
+		// when day is like "Monday" convert to number
+		if ( in_array($meeting_day_input, $tsml_days) ) {
+			$dow_number = date('N', strtotime( $meeting_day_input ) );
+			if ($meeting_day_input == 'Sunday') {$dow_number = '0';}
```

Comment:
sooooo, this would be Sunday = 0, Monday = 2, Tuesday = 3. instead of these lines you could use
```
$dow_number = date('w', strtotime($meeting_day_input));
```
or you should be able to do 
```
$dow_number = array_search($meeting_day_input, $tsml_days);
```

Reply:
I was wondering why **$dow_number = date('N', strtotime( $meeting_day_input ) );** returned 1 thru 6 for Monday to Saturday, and 7 for Sunday. I was hoping I could find something better than my hack. I'll give your suggestion a shot tomorrow.  Thanks for the help.

Reply:
array_search worked like a charm. No more hack...